### DISCUSSION TITLE
New variable in form state

### DISCUSSION THREAD

**Discussion by anshour** [react-hook-form/react-hook-form#8967] [Ideas] [answered]
Hello, i think it would be great to add isLoading variable in the formState object. this variable is combination from isValidating and isSubmitting, so we can prevent user from changing any input if isLoading is true, and we can provide faster feedback to user if the submit button clicked (normally, the loading indicator will show when isSubmitting true because isValidating happen so fast, but in the certain condition, isValidating is taking some time, this cause user click submit button twice or more)

**Comment by bluebill1049** [comment]
Could you use `isSubmitting` to disable the submit button.

**Reply by anshour** [reply]
yes i could, but i also want to disable the button when isValidating is true (i know i could simply type && logic, but i think its faster to type isLoading)

**Comment by Moshyfawn** [comment, marked as answer]
I don't think the combined `boolean` is with it to extend the RHF API surface like that

**Reply by Moshyfawn** [reply]
Additionally, with the `formState` subscription model, it's better to keep all states as separate variables to subscribe to selectively
